# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/datamodel/FileSystemDataManager.h
# Commit: cfa7bbd206c3
# Full Hash: cfa7bbd206c3141f7d0b2801586dec142f20a11e
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 2
# Description:
#   Bug 1786465 - Create only one FileSystemDataManager per origin; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155553
# ==============================================================================

diff -r a2f6102fc8ea -r cfa7bbd206c3 dom/fs/parent/datamodel/FileSystemDataManager.h
--- a/dom/fs/parent/datamodel/FileSystemDataManager.h	Wed Aug 31 09:34:55 2022 +0000
+++ b/dom/fs/parent/datamodel/FileSystemDataManager.h	Wed Aug 31 09:34:55 2022 +0000
@@ -9,8 +9,10 @@
 
 #include "mozilla/NotNull.h"
 #include "mozilla/dom/FileSystemTypes.h"
+#include "mozilla/dom/quota/CheckedUnsafePtr.h"
 #include "nsCOMPtr.h"
 #include "nsISupportsUtils.h"
+#include "nsString.h"
 
 namespace mozilla {
 
@@ -19,13 +21,14 @@
 
 namespace dom::fs::data {
 
-class FileSystemDataManager {
+class FileSystemDataManager
+    : public SupportsCheckedUnsafePtr<CheckIf<DiagnosticAssertEnabled>> {
  public:
-  FileSystemDataManager();
+  explicit FileSystemDataManager(const Origin& aOrigin);
 
   using result_t = Result<RefPtr<FileSystemDataManager>, nsresult>;
-  static FileSystemDataManager::result_t CreateFileSystemDataManager(
-      const fs::Origin& aOrigin);
+  static FileSystemDataManager::result_t GetOrCreateFileSystemDataManager(
+      const Origin& aOrigin);
 
   NS_INLINE_DECL_REFCOUNTING(FileSystemDataManager)
 
@@ -34,8 +37,9 @@
   }
 
  protected:
-  ~FileSystemDataManager() = default;
+  ~FileSystemDataManager();
 
+  const Origin mOrigin;
   const NotNull<nsCOMPtr<nsISerialEventTarget>> mBackgroundTarget;
 };
 
